src/App.tsx

```diff
@@ -15,36 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const CLASS_IS_LIGHT = 'is-light';
+
+enum SortType {
+  BY_ALPHABET = 'alphabet',
+  BY_LENGTH = 'length',
+  BY_DEFAULT = '',
+}
+
+type GetPreparedGoods = (
+  goods: string[],
+  sortField: string,
+  reversed: boolean,
+) => string[];
+
+const getPreparedGoods: GetPreparedGoods = (goods, sortField, reversed) => {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.BY_ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.BY_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.BY_DEFAULT);
+  const [reversed, isReversed] = useState(false);
```

Comment:
Boolean variables should start from `is` or `has`. And useState method should start from `set`
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```